### ISSUE TITLE
PHP Extensions

### ISSUE THREAD

**Issue by mattheath** [mattheath/puppet-php#6]
Need a way to manage compiled php extensions. homebrew-php has a huge number cataloged, but these would be difficult to port over as I want to support multiple versions of PHP (eg. multiple minor versions of 5.4). Currently experimenting with a custom type along the same vein as the ruby module's rubygem provider in the extensions branch.

The current solution will mean that an extension would have to be manually defined for each version of PHP installed, which isn't ideal if there are a large number of extensions required for each version. Maybe an extension 'set' type?

**Comment by mattheath** [comment]
Custom type merged in for `0.3.0`. Currently the syntax below is used. I've added a few extensions from pecl, but either need to go through and add all of them, or refactor sufficiently that a generic solution can be achieved without manually defining them all.

``` puppet
php::extension::apc { "apc for ${version}":
  php     => $version,
  version => '3.1.13', # Optionally specify the extension version
}
```

**Comment by mattheath** [comment]
Added a git provider to complement the pecl provider - this compiles an extension from a cached git repo. An example is the zmq extension.

Another issue with the current management method is that a php service won't be refreshed if an extension is added or changed.
